src/actions/slice.js

```diff
@@ -0,0 +1,28 @@
+export const APPLY_SLICE = 'APPLY_SLICE';
+
+export const applySlice = (apply) => {
+  return async function (dispatch) {
+    dispatch({
+      type: APPLY_SLICE,
+      apply,
+    });
+  };
+};
+
+export const SLICE_PIPELINE = 'SLICE_PIPELINE';
```

Comment:
nit, but maybe we can have consistent naming 

```
export const APPLY_SLICE = 'APPLY_SLICE';
export const SET_SLICE = 'SET_SLICE';
export const RESET_SLICE = 'RESET_SLICE';
```

Reply:
hey good suggestion, I've updated to be consistent 

export const APPLY_SLICE_PIPELINE = 'APPLY_SLICE_PIPELINE';
export const SET_SLICE_PIPELINE  = 'SET_SLICE_PIPELINE ';
export const RESET_SLICE _PIPELINE = 'RESET_SLICE_PIPELINE ';